### PR TITLE
UI Modernization: Move Me to bottom tab bar

### LINKED ISSUE #21273: Me – Move Me button to the tab bar item in Jetpack
- [x] Move the "Me" button to the tab bar
- [x] Remove the top padding on the My Site screen
- [x] Make sure the new "Me" menu has no back button

Scope:
- Only the scenario where the user has one or more sites and the tab bar is present

### LINKED ISSUE #21283: Me – Show Me item in the site menu in WordPress

### PR DESCRIPTION
**Pull request by momo-ozawa**
Closes #21273
Closes #21283 

## Description

- Moves Me to bottom tab bar
    - Displays Me in split view, reverse some changes made in https://github.com/wordpress-mobile/WordPress-iOS/pull/15042/files 
- Removes top padding from My Site screen
- Adds Me menu item for simplified WPiOS Ui

### Updated Me location

<details open>
<summary>Show expected Me location for WP</summary>
<br>

Jetpack Feature Removal Phase | UI type mapping | Expected Me location | iPhone | iPad
-- | -- | -- | -- | --
.staticScreens (current phase) | .staticScreens | bottom tab bar |   | 
.four, .newUsers, .selfHosted | .simplified | menu |  | 

</details>

- On WP, change the feature removal phase via the debug menu and verify it matches the expected Me location for bot h iPhone and iPad (expand table above)
- On JP, verify Me has moved down to the bottom tab bar for both iPhone and iPad (feature removal phases are irrelevant for JP - Me will always be shown in the bottom tab bar)

### Me icon updates

- On WP, change the removal phase to static screens via the debug menu
- Force quit then reopen
- Update the profile icon
- Verify the me tab icon changes to the new icon
- Change the removal phase to phase four
- Force quit then reopen
- Verify the me menu item changes to the new icon

### Removed top padding

<details open>
<summary>Show before/after</summary>
<br>

Before | After
-- | --
 | 

</details>

- Verify the top padding above the My Site header is no longer there
- Verify the title appears on scroll

## Regression Notes
1. Potential unintended areas of impact
Me tab

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by kean** [comment]
JP: works great; updates when you change the screenshot within the app
WP: couldn't find the Me button: [IMG_C15C9B1EFEB2-1](https://github.com/wordpress-mobile/WordPress-iOS/assets/1567433/0f7e96f5-50b4-49cf-9486-d73b9220cc40)

_Update_: I re-rested WP and, I'm not sure why, but I can now see the "Me" tab under "Configure".

File: WordPress/Classes/Utility/Media/ImageDownloader+Gravatar.swift [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+                    completion(nil)
+                    return
+                }
+
+                completion(image)
+            }
+        }
+    }
+
+    private func gravatarUrl(for email: String) -> URL? {
```
Comment (kean) [inline WordPress/Classes/Utility/Media/ImageDownloader+Gravatar.swift]: There should be some existing code for creating the URLs, potentially this: `downloadGravatarWithEmail`.

File: WordPress/Classes/ViewRelated/Me/Me Main/MeViewController.swift [hunk 2]
```diff
@@ -485,6 +508,19 @@ class MeViewController: UITableViewController {
     }()
 }
 
+// MARK: - WPSplitViewControllerDetailProvider Conformance
+
+extension MeViewController: WPSplitViewControllerDetailProvider {
+    func initialDetailViewControllerForSplitView(_ splitView: WPSplitViewController) -> UIViewController? {
+        // If we're not logged in yet, return app settings
+        guard let _ = defaultAccount() else {
+            return AppSettingsViewController()
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Me/Me Main/MeViewController.swift]: Is it is valid scenario? Is there a way to reproduce it?

Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Me/Me Main/MeViewController.swift]: I don't think so, good catch. Will remove this

File: WordPress/Classes/Utility/Media/ImageDownloader+Gravatar.swift (hunk 1 above)
Comment (momo-ozawa) [reply WordPress/Classes/Utility/Media/ImageDownloader+Gravatar.swift]: As discussed, I've moved over this logic to `Gravatar.swift` 🙇‍♀️ 
https://github.com/wordpress-mobile/WordPressUI-iOS/pull/133

File: WordPress/Classes/ViewRelated/Me/Me Main/MeViewController.swift (hunk 2 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Me/Me Main/MeViewController.swift]: Done [ebe5adb](https://github.com/wordpress-mobile/WordPress-iOS/pull/21348/commits/ebe5adb0364deda4cb4ca541164e50fc4acba214)

If the default account isn't set (i.e. after you've logged out), returning nil - this prevents the app from crashing on logout

File: WordPress/Classes/Extensions/UIImage+Gravatar.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            let ovalPath = UIBezierPath(ovalIn: CGRect(origin: .zero, size: imageSize))
+            ovalPath.addClip()
+            draw(at: CGPoint(x: -xOffset, y: -yOffset))
+        }
+    }
+}
+
+extension UITabBarItem {
+
+    func updateGravatarImage(_ image: UIImage) {
```
Comment (kean) [inline WordPress/Classes/Extensions/UIImage+Gravatar.swift]: Unused code.

File: WordPress/Classes/Utility/Media/ImageDownloader+Gravatar.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+extension ImageDownloader {
+
+    func downloadGravatarImage(with email: String, completion: @escaping (UIImage?) -> Void) {
+
+        guard let url = Gravatar.gravatarUrl(for: email) else {
+            completion(nil)
+            return
+        }
+
+        if let cachedImage = ImageCache.shared.getImage(forKey: url.absoluteString) {
```
Comment (kean) [inline WordPress/Classes/Utility/Media/ImageDownloader+Gravatar.swift:12]: It's probably not ideal to cache avatars in memory without providing a time-to-live interval that roughly matches the HTTP cache control headers (around 5 minutes?). If you change the outside of the app, it won't update until the memory cache clears, which may need to happen while the app is running.

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Me.swift [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        }
+
+        ImageDownloader.shared.downloadGravatarImage(with: email) { [weak self] image in
+            guard let image,
+                  let gravatarIcon = image.gravatarIcon() else {
+                return
+            }
+
+            row.image = gravatarIcon
+            self?.tableView.reloadData()  // FIXME: only reload Me row
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Me.swift]: Will it be addressed in the scope of this PR?

**Comment by kean** [comment]
Re-tested the flow. There are too many migration phases lol!
Approved, but I think we need to add TTL to [the cache](https://github.com/wordpress-mobile/WordPress-iOS/pull/21348#discussion_r1303657616), but maybe after it's merged and globally.

**Review by kean — APPROVED** [review]

File: WordPress/Classes/Utility/Media/ImageDownloader+Gravatar.swift (hunk 4 above)
Comment (momo-ozawa) [reply WordPress/Classes/Utility/Media/ImageDownloader+Gravatar.swift:12]: Good point! Will address this in a separate PR

**Comment by osullivanchris** [comment]
This looks good. Some minor comments/ideas:
- I noticed (was probably always this way) that in dark mode, on 'Me' my username is a dark grey, rather than white(primary label color)
- Do you also feel that the site icon on My Site looks too large after this change? It wasn't noticeable before, but now I feel it is.
- As mentioned on slack, I do think the user image needs to have some transparency when not selected. On Android we used 36% which I found was like a standard opacity value somewhere in Material. Could use same, unless you have some system value for iOS.
- The profile image looks a little small. Looks like it is 24pt right now. Other icons I think are 32pt, but have some internal padding. I think 28pt for the profile image in the tab bar will be optically about right.
- My own image is blurry in the tab bar. Not sure why. Looks fine in the larger 'Me' image.

**Comment by momo-ozawa** [comment]
Hey @osullivanchris, thanks for reviewing! I'll address your comments in a separate PR 🙇‍♀️ 

> I noticed (was probably always this way) that in dark mode, on 'Me' my username is a dark grey, rather than white(primary label color)

I can change the @username to white in dark mode. Does light mode look okay?

Screenshot 2023-08-24 at 16 20 39 | Screenshot 2023-08-24 at 16 21 32
-- | --

> Do you also feel that the site icon on My Site looks too large after this change? It wasn't noticeable before, but now I feel it is.

Actually now that you mention it... yeah, it looks a bit big. I can bump it down a few pts and we can see what looks good

> As mentioned on slack, I do think the user image needs to have some transparency when not selected

Noted!

> 28pt for the profile image in the tab bar will be optically about right

Sounds good. Should I also bump the menu item icon to 28 pt?

> My own image is blurry in the tab bar.

Hmm! Will investigate!

**Comment by osullivanchris** [comment]
> I can change the @username to white in dark mode. Does light mode look okay?

oh yeah! It is ever so slightly grey in light mode as well. Primary label color (AKA black!) would be better.

> Actually now that you mention it... yeah, it looks a bit big. I can bump it down a few pts and we can see what looks good

sounds good. It looks like its 64 right now. I have gone all the way down to 40/48 and it looks pretty good. It might mean a smaller max type size though. As the type could be taller than the icon otherwise. Have a play around with it and let me know what you think.

> Sounds good. Should I also bump the menu item icon to 28 pt?

Which menu icon, sorry?

**Comment by momo-ozawa** [comment]
> Which menu icon

@osullivanchris This one! When the Me icon shows up in the menu. Should I bump this up from 24 to 28?

**Comment by osullivanchris** [comment]
> @osullivanchris This one! When the Me icon shows up in the menu. Should I bump this up from 24 to 28?

Aha. No that looks good as it is 👍

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Me.swift (hunk 5 above)
Comment (momo-ozawa) [reply WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController+Me.swift]: Yes! https://github.com/wordpress-mobile/WordPress-iOS/pull/21348/commits/f4db05433065e6ec9fc11455ee39143e1938dc31

**Pull request merged**